### PR TITLE
EARTH-18 Link banner tweeks

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Fixed rendering with fields
- Used `image` instead of `imgsrc` for region key

# Needed By (Date)
- End of sprint

# Urgency
- Some kinda

# Steps to Test

1. Checkout `EARTH-18-link-banner` branch of [stanford_paragraph_types](https://github.com/SU-SWS/stanford_paragraph_types/pull/44)
2. Checkout this branch.
3. `drush en stanford_link_banner -y`
4. Ensure patterns/link_banner page still looks correct
5. Create a complex page with a "Link Banner" paragraph item.
6. Ensure it looks about correct.

# Associated Issues and/or People
- https://github.com/SU-SWS/stanford_paragraph_types/pull/44

# See Also
- [stanford_paragraph_types](https://github.com/SU-SWS/stanford_paragraph_types/pull/44)

### CODE REVIEW

File: patterns/molecules/link-banner/link-banner.ui_patterns.yml [hunk 1]
```diff
@@ -9,10 +9,10 @@ link_banner:
       label: Description
       preview: "Find our research and insights, organized into nine categories, at our Earth Matters research news magazine."
       type: text
-    imgurl:
+    iamge:
```
Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.ui_patterns.yml]: typo.

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.ui_patterns.yml]: holy speedy

File: patterns/molecules/link-banner/link-banner.html.twig [hunk 2]
```diff
@@ -5,26 +5,27 @@
...
-      }
-    %}
-
-    {# Pass vars through to include template. #}
-    {% include '@stanford_components/atoms/link-item/link-item.html.twig' with vars %}
-  {% endfor %}
+        {# Pass vars through to include template. #}
+        {% include '@stanford_components/atoms/link-item/link-item.html.twig' with vars %}
+      {% else %}
+        {{ item|render|striptags('<a><span>')|trim|raw }}
```
Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.html.twig]: Can you avoid raw?

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: I wanted to. its either all that or we can change some styles. but not sure if the markup changes with entities vs views, vs panels etc.

Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.html.twig]: What is the difference between
`item|render|striptags('<a><span>')|trim|raw` 
and
`item|render|striptags('<a><span>')|trim`
?

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: it seems that anytime we want to retain any tags after the `render` we need to use raw :(

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: it sanitizes it and shows the markup to the user

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: https://www.drupal.org/node/2705249

Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.html.twig]: Huh, interesting. So render performs the sanitization and XSS filtering before we use striptags and then output as raw. Gotcha. Can you leave an inline comment above this so the next guy groks it as well.

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: 👍 You got it.

Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.html.twig]: 👍

File: patterns/molecules/link-banner/link-banner.html.twig [hunk 3]
```diff
@@ -5,26 +5,28 @@
  */
 #}
 {{ attach_library('ui_patterns/link_item.link_item_library') }}
-<div class="link-banner" style="background-image:url({{ imgurl|default('https://upload.wikimedia.org/wikipedia/commons/3/36/Hopetoun_falls.jpg') }})">
+<div {{ attributes.addClass('link-banner') }} style="background-image:url({{ image|get_img_src }})">
```
Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.html.twig]: The background image is still optional even though a default image is no longer provided. Can you do a check for a value before adding the background style.

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: I could add the default back if that would work. this is the only place i've seen the default specified.

Comment (sherakama) [inline patterns/molecules/link-banner/link-banner.html.twig]: Keep the default out of the twig file. There is a possibility to have a solid color background instead.

Comment (pookmish) [inline patterns/molecules/link-banner/link-banner.html.twig]: 👍 💥

**Pull request merged**